drafts/signals-everywhere.md

```diff
@@ -0,0 +1,167 @@
+---
+title: Signals, Signals everywhere
+subtitle: "An explainer in the dawn of a new era as Web Frameworks converge"
+domain: software-engineering-corner.zuehlke.com
+tags: signals, web-development, javascript, frontend-development, reactivity, explained
+cover: https://cdn.hashnode.com/res/hashnode/image/upload/v1708268993236/6Qa2m8luN.jpg?auto=format
+publishAs: culas
+hideFromHashnodeCommunity: false
+saveAsDraft: true
+---
+
+After decades of JavaScript Wild West exploration with countless frameworks and libraries left in the dust, the recent emergence of Signals is the best thing to happen since ES5.
+If you're unsure about what Signals are and why they're important, it's time to learn.
+**Every savvy web developer should know and understand Signals** to create interactive experiences on the web.
+
+## The Problem
+
+All JavaScript frameworks essentially try to solve one single challenge at their core: it's all about **"keeping the UI in sync with the state"** (see [The deepest reason why modern JavaScript frameworks exist](https://medium.com/dailyjs/the-deepest-reason-why-modern-javascript-frameworks-exist-933b86ebc445)).
+One way or another, we need to update our visual interface every time the state changes, something React describes as `UI = fn(state)`.
+
+Vanilla JavaScript only gives us rudimentary, imperative APIs to achieve that.
+The complexity of an interactive website rapidly grows beyond what is reasonably maintainable without the use of third-party libraries and frameworks.
+They can provide structured ways to declaratively define our views, and manage state, and we can leave the synching itself to the frameworks.
+Each new framework proposed a new or slightly different way to make our UI reactive, while still providing good developer ergonomics (or "developer experience", DX).
+Angular uses a conservative change detection algorithm, while React introduced a virtual DOM for better performance.
+And then there's Svelte, which uses a compiler to analyse dependencies at build time.
```

Comment:
i would merge these two sentences:
Angular usese...., React introduced the virtual.... and Svelte analyse dependencies through compiler....

something like that :D